lldb/source/Interpreter/CommandInterpreter.cpp

```diff
@@ -3306,6 +3306,8 @@ bool CommandInterpreter::SaveTranscript(
   result.SetStatus(eReturnStatusSuccessFinishNoResult);
   result.AppendMessageWithFormat("Session's transcripts saved to %s\n",
                                  output_file->c_str());
+  if (!GetSaveTranscript())
+    result.AppendError("Note: the setting interpreter.save-transcript is set to false, so the transcript might not have been recorded.");
```

Comment:
We should do this at the beginning of the function, before doing the work and return early if `save-transcript` is disabled.

Reply:
Sounds good to me. I'll make the change.

Reply:
See my [comment above](https://github.com/llvm/llvm-project/pull/109020#discussion_r1769375600).

I hate to be so pedantic about it, but I would be worried if users used this flow of 
```
(lldb) settings set interpreter.save-transcript true
... do various things they want to save ...
(lldb) settings set interpreter.save-transcript false
... do various things they don't want to save ...
(lldb) session save 
```
This wouldn't work anymore if we early returned.

To your point though, I guess they could always re-toggle it before that last session save, and maybe it would be better for the average user.

Anyway, that's why I decided to make this commit so conservative.

Reply:
@medismailben friendly ping on any followup comments. If not, I'll go ahead and merge.